provider_shopper/lib/models/cart.dart

```diff
@@ -7,7 +7,7 @@ import 'package:provider_shopper/models/catalog.dart';
 
 class CartModel extends ChangeNotifier {
   /// The private field backing [catalog].
-  CatalogModel _catalog;
+  late CatalogModel _catalog;
```

Comment:
Consider using `CatalogModel?` instead of `late CatalogModel` here. It makes the code more verbose but we can be more sure we're not unwittingly letting some null error slip through.

If we were _sure_ that `_catalog` will be set before it's used (e.g. it's assigned in the Constructor), then `late` is a great option.

Reply:
Yes, making it `CatelogModel?` makes complete sense. But, since I am changing this, I think I also need to change its getters and setters. For example :
`CatalogModel? get catalog => _catalog;`

Reply:
Yes, that change will reverberate through the code.